src/librustc_trans/back/link.rs

```diff
@@ -938,8 +938,10 @@ fn add_upstream_rust_crates(cmd: &mut Linker,
             Linkage::NotLinked |
             Linkage::IncludedFromDylib => {}
             Linkage::Static => {
+                let is_a_no_builtins_crate =
+                    attr::contains_name(&sess.cstore.crate_attrs(cnum), "no_builtins");
```

Comment:
Could this logic for checking this attribute be centralized somewhere? I think there's probably at least one other instance of this check (besides the two added here) which reads that in the first place as well.